app/src/main/java/org/jellyfin/androidtv/util/profile/ProfileHelper.kt

```diff
@@ -63,6 +57,68 @@ object ProfileHelper {
 		}
 	}
 
+	val deviceAVCCodecProfile by lazy {
+		buildList {
+			when {
+				!MediaTest.supportsAVC() -> {
+					Timber.i("*** Does NOT support AVC")
+					add(
+						ProfileCondition(
+							ProfileConditionType.Equals,
+							ProfileConditionValue.VideoProfile,
+							"none"
+						)
+					)
+				}
+				!MediaTest.supportsAVCHigh10() -> {
+					Timber.i("*** Does NOT support AVC High 10 bit")
+					add(
+						ProfileCondition(
+							ProfileConditionType.NotEquals,
+							ProfileConditionValue.VideoProfile,
+							"High 10"
+						)
+					)
+				}
+				else -> {
+					Timber.i("*** Supports AVC High 10 bit")
+					add(
+						ProfileCondition(
+							ProfileConditionType.NotEquals,
+							ProfileConditionValue.VideoProfile,
+							"none"
+						)
+					)
+				}
+			}
+		}
+	}
+
+	val deviceAVCLevelCodecProfiles by lazy {
+		buildList {
+			when {
+				MediaTest.supportsAVCHigh10() -> {
+					add(
+						ProfileCondition(
+							ProfileConditionType.LessThanEqual,
+							ProfileConditionValue.VideoLevel,
+							MediaTest.getAVCHigh10Level()
```

Comment:
This is wrong because you're constraining the level of _all_ AVC profiles when "high 10" is supported by the limit of "high 10". That means if a decoder supports a lower level of "high 10" than other profiles, you're going to be unnecessarily transcoding those other profiles.

Consider the case of `getAVCHigh10Level()` returning "40" and `getAVCMainLevel()` returning "52". You'll end up transcoding all AVC "main" and "high" streams above level 4.0, when the decoder handles up to 5.2.